### PR TITLE
Remove obsolete check for test mode in failhard

### PR DESCRIPTION
**Pull request by aneeshusa**
There is an explicit check above to ignore failhard in test=True mode,
so this check to continue on results of `None` is obsolete.

### What does this PR do?
Remove an obsolete check.

### What issues does this PR fix or reference?
Essentially reverts #36078 as #36199 was merged later.
Discussion on the issue #18341.

### Previous Behavior
Failhard is ignored in test=True mode.

### New Behavior
Unchanged

### Tests written?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

**Comment by aneeshusa** [comment]
Actually, I wonder how this interacts with `prereq` and other cases where a single state is running with `test=True`, but not the state run as a whole. (I was reminded by [this comment](https://github.com/saltstack/salt/issues/42553#issuecomment-322594128) by @whiteinge).

**Review by thatch45 — APPROVED** [review]

**Comment by aneeshusa** [comment]
I'd like to get a chance to test this locally (and/or add a test case for that scenario with `prereq`) before this is merged, so please hold off for now.

**Comment by thatch45** [comment]
Sounds good, thanks!

**Comment by cachedout** [comment]
Hi @aneeshusa. Any update on testing progress here?

**Comment by cachedout** [comment]
Hi @aneeshusa. Any update on testing progress here?

**Comment by aneeshusa** [comment]
This is a lower priority PR for me and I'm going to close it for now as it's on my back burner. I'll re-open this if I find time to test this out.

**Comment by cachedout** [comment]
@aneeshusa I appreciate that. Thank you!

**Pull request closed without merging**
